### ISSUE TITLE
Support problem matching for concurrent prefixed output, e.g. concurrently, npm-run-all, etc.

### ISSUE THREAD

**Issue by segevfiner** [microsoft/vscode#158821]
When working on monorepos it is common to need to run multiple watch/build commands in parallel using tools such as `pnpm -r`, `concurrently`, `npm-run-all`, `nx`, `turborepo`. Sometimes even in nested invocations, such as `pnpm --filter` from `npm-run-all`. Such tools often prefix each command's output which prevents using problem matchers with such setups.

Not only is the parallel output likely to confuse problem matching as it might match the output of parallel commands and pair them incorrectly (Watch begin/end patterns), but it will likely won't match at all as many/most problem matchers anchor their match to the start of the line for accuracy.

It would be nice to have some way to still use problem matchers in this scenario as they are quite convenient, being able to see at a glance if a background watch task errored and jumping to the error.

Note that in such a configuration you have multiple watchers firing of at random and finishing at random with interleaved output which is often prefixed, but some tools even allow to have them unprefixed.

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by VSCodeTriageBot** [comment]
This feature request has not yet received the 20 community [upvotes](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) it takes to make to our backlog. 10 days to go. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by VSCodeTriageBot** [comment]
:slightly_frowning_face: In the last 60 days, this feature request has received less than 20 community upvotes and we closed it. Still a big Thank You to you for taking the time to create this issue! To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by martaver** [comment]
I think this is a great idea, especially relevant since there is a big push towards monorepos these days. This issue should stay open and find a slot in the backlog, it would be great to be able to leverage 'Problems' in monorepo set ups that make incremental rebuilds possible, such as turborepo.
